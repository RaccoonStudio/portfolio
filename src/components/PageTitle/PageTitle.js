import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { SizingSystem, TypographySystem } from "../core"

const StyledPageTitle = styled.h1`
  margin-bottom: 0.3em;
  font-size: 3.2rem;
  font-weight: ${TypographySystem.weights.light};

  @media (min-width: ${SizingSystem.media.laptop}) {
    font-size: 4rem;
    font-weight: ${TypographySystem.weights.thin};
    text-align: center;
  }

  @media (min-width: ${SizingSystem.media.desktop}) {
    font-size: 4.8rem;
    font-weight: ${TypographySystem.weights.hairstyle};
  }
`

const PageTitle = props => (
  <>
    <StyledPageTitle children={props.children} {...props} />
  </>
)

PageTitle.propTypes = {
  /**
   * Specify the content of the title
   */
  children: PropTypes.node.isRequired,
}

PageTitle.defaultProps = {}

export default PageTitle
