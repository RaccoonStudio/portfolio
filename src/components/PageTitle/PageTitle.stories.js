import React from 'react';

import { storiesOf } from '@storybook/react';
import PageTitle from './PageTitle';

storiesOf('PageTitle', module)
  .add('overview', () => <PageTitle>Hello World!</PageTitle>);
