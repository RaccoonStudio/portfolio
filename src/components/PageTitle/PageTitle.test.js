import React from "react"
import renderer from "react-test-renderer"

import PageTitle from "./PageTitle"

describe("PageTitle", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<PageTitle>Hello World!</PageTitle>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
