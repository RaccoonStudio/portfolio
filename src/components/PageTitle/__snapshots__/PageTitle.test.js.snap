// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders correctly 1`] = `
<h1
  className="sc-bdVaJa iTWoZQ"
>
  Hello World!
</h1>
`;
